### DISCUSSION TITLE
how it works "dbus-serialbattery/conf/serial-starter.d" ?

### DISCUSSION THREAD

**Discussion by stanhausc** [Louisvdw/dbus-serialbattery#104] [Q&A] [answered]
Hello, @Louisvdw  and community, could you please explain these lines from the serial-starter.d file?

service sbattery dbus-serialbattery
default alias gps:vedirect:sbattery
alias rs485 cgwacs:fzsonick:imt:modbus:sbattery

I ask because we have fzsonick batteries, and the driver for them was developed recently, I want to understand this logic, what happens when the serial-starter.d file is called and what does the following line do?

  "alias rs485 cgwacs:fzsonick:imt:modbus:sbattery"

Is this line used in your drive?

On the VIctron website, the batteries officially supported use the CAN protocol, however, those in your drive and Fzsonick use Modbus. 

![Fzsonick](https://user-images.githubusercontent.com/85182432/157520220-6d28c381-2e75-4190-8fe5-04d7352cdc4d.jpeg)

**Comment by Louisvdw** [comment, marked as answer]
You get 2 types of drivers that run on the VenusOS devices. Those that always run, and those that depend an a serial connection. 

For serial connection devices there is a service that checks for any new connections, and then will start all the available drivers until a driver "sticks" to a serial port. That service is called **serial-starter** and it has a config file at /etc/venus/serial-starter.conf

The default one will look like this. The first few lines lists all the services, their short names and the folder/driver name.
The alias lines tells which services should be tested when a new RS485 or RS232/TTL (default) connection is found. (I think CAN will also fall under RS485)
```
service cgwacs          dbus-cgwacs
service gps             gps-dbus
service modbus          dbus-modbus-client.serial
service modem           dbus-modem
service mkx             mk2-dbus
service vedirect        vedirect-interface
service fzsonick        dbus-fzsonick-48tl
service imt             dbus-imt-si-rs485tc

alias   rs485           cgwacs:fzsonick:imt:modbus
alias   default         gps:vedirect

include /data/conf/serial-starter.d
```
This last line is to include an extra config file (the one I use) that does not get distroyed and is not part of the VenusOS install.

So you should see that my serial-starter.d should make more sense now.
1. I add a new service called sbattery
2. and then adjust the alias to include sbattery to be tested for both RS485 and TTL connections.
3. The alias order is also the order in which the drivers will be tested.

You can fine more details at https://github.com/victronenergy/venus/wiki/howto-add-a-driver-to-Venus#3-installing-a-driver
